hashes/src/sha256.rs

```diff
@@ -180,53 +214,39 @@ impl Midstate {
     }
 }
 
-impl fmt::LowerHex for Midstate {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        hex::fmt_hex_exact!(f, 32, self.0.iter().rev(), hex::Case::Lower)
-    }
-}
-
-impl fmt::UpperHex for Midstate {
+impl fmt::Debug for Midstate {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        hex::fmt_hex_exact!(f, 32, self.0.iter().rev(), hex::Case::Upper)
+        f.debug_struct("Midstate")
+            .field("bytes", &self.bytes.as_hex())
+            .field("length", &self.length)
+            .finish()
     }
 }
 
-impl fmt::Display for Midstate {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { fmt::LowerHex::fmt(&self, f) }
-}
-
-impl fmt::Debug for Midstate {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { write!(f, "0x{}", self.0.as_hex()) }
-}
-
 impl convert::AsRef<[u8]> for Midstate {
-    fn as_ref(&self) -> &[u8] { &self.0 }
+    fn as_ref(&self) -> &[u8] { &self.bytes }
 }
 
-impl<I: SliceIndex<[u8]>> Index<I> for Midstate {
-    type Output = I::Output;
-
-    #[inline]
-    fn index(&self, index: I) -> &Self::Output { &self.0[index] }
+/// `bytes_hashed` invariant violated (not a multiple of 64).
```

Comment:
Do you mean `Midstate` invariant violated?

Reply:
Yes, "`Midstate` invariant ..." reads better, thanks.